### ISSUE TITLE
plt.stackplot() does not work with 3.0.0

### ISSUE THREAD

**Issue by soerendip** [matplotlib/matplotlib#12405]
### Bug report
Something now breaks this code that used to work. Now, I have to call plt.subplots first and use stackplot on an axis object. 

TypeError: stackplot() got multiple values for argument 'x'

```python
from matplotlib import pyplot as plt
plt.stackplot([1,2,3], [1,2,3]) 
```

Should be equivalent to 
```python

fig, ax = plt.subplots()
ax.stackplot([1,2,3], [1,2,3]) 
```

**Matplotlib version**

  * Operating system:
  * Matplotlib version: 3.0.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): 'module://ipykernel.pylab.backend_inline'
  * Python version: 3.6.6
  * Jupyter version (if applicable): 4.4.0
  * Other libraries: 

Installed matplotlib using conda with default channel.

**Comment by jklymak** [comment]
This is present in master as well.

**Comment by jklymak** [comment]
Bisects to 86fe16bfe7 https://github.com/matplotlib/matplotlib/pull/10918

**Comment by ImportanceOfBeingErnest** [comment]
But only when run in a notebook, not when run as script.

**Comment by anntzer** [comment]
#12293 also closes this.

**Comment by timhoffm** [comment]
Closed via #12293.
